File: python310/task_1.py
import random


class RandomArray:
    def __init__(self, array_length):
        if isinstance(array_length, int):
            self.array_length = array_length
            self._gen_new_array()
        else:
            raise ValueError('array_length is not an integer')

    def _gen_new_array(self):
        self.array = [random.random() for _ in range(self.array_length)]

    def get_new_array(self):
        self._gen_new_array()
        return self.array

    def get_array(self):
        return self.array

    def get_min(self):
        return min(self.array)

    def get_max(self):
        return max(self.array)


if __name__ == "__main__":
    flag = True
    while flag:
        array_length = input("Длинна массива: ")
        if isinstance(int(array_length), int):
            flag = False
            array_length = int(array_length)
        else:
            print('length is not an integer')
    new_array = RandomArray(array_length)
    print(f'Массив: {new_array.get_array()}')
    print(f'Минимальный элемент: {new_array.get_min()}')
    print(f'Максимальный элемент: {new_array.get_max()}')

File: JS/task_1.js
export const task1 = 'Задача 1. Сформировать и заполнить массив случайным числами и вывести максимальное, ' +
    'минимальное и среднее значение. Для генерации случайного числа использовать метод Math.random(), ' +
    'который возвращает значение в промежутке [0, 1].'

export class RandomArray {
    constructor(arrayLength) {
        if (typeof arrayLength === 'number' && Number.isInteger(arrayLength)) {
            this.arrayLength = arrayLength;
            this._genNewArray();
        } else {
            throw new Error('arrayLength is not an integer');
        }
    }

    _genNewArray() {
        this.array = Array.from({ length: this.arrayLength }, () => Math.random());
    }

    getNewArray() {
        this._genNewArray();
        return this.array;
    }

    getArray() {
        return this.array;
    }

    getMin() {
        return Math.min(...this.array);
    }

    getAvg() {
        return this.array.reduce((sum, current) => sum + current, 0)/this.array.length;
    }

    getMax() {
        return Math.max(...this.array);
    }
}

// Пример использования
// const randomArray = new RandomArray(10); // Создать случайный массив длиной 10
// console.log(randomArray.getArray()); // Вывести массив
// console.log(randomArray.getMin()); // Вывести минимальное значение
// console.log(randomArray.getMax()); // Вывести максимальное значение